src/Microsoft.ServiceFabric.Services.Remoting/Description/InterfaceDescription.cs

```diff
@@ -67,10 +67,13 @@ public MethodDescription[] Methods
             MethodReturnCheck methodReturnCheck)
         {
             EnsureValidMethods(remotedInterfaceKindName, remotedInterfaceType, methodReturnCheck);
-            var methods = remotedInterfaceType.GetMethods().Select(
-                methodInfo => MethodDescription.Create(remotedInterfaceKindName, methodInfo)).ToList();
-
-            return methods.ToArray();
+            var methods = remotedInterfaceType.GetMethods();
```

Comment:
I agree with #4, ToList() and then ToArray() is redundant. It can just be ToArray(), IL code is doing it. #1 and #2 are fine. 
So Its just #3 which causes list to be resized as IEnumerable returned from Select is not ICollection? If yes, then lets make the change.